drivers/adc/adc_npcx.c

```diff
@@ -296,6 +402,268 @@ static int adc_npcx_read_async(const struct device *dev,
 }
 #endif /* CONFIG_ADC_ASYNC */
 
+static void adc_npcx_set_repetitive(const struct device *dev, int chnsel,
```

Comment:
Refer to this this [comment](https://github.com/zephyrproject-rtos/zephyr/pull/41515#discussion_r802033318).

Reply:
Please verify these functions are not included if `CONFIG_ADC_CMP_NPCX=n`, otherwise please add a guard.

Reply:
> Please verify these functions are not included if `CONFIG_ADC_CMP_NPCX=n`, otherwise please add a guard.

I created this PR: #42807, to test how disabling optimizations will impact footprint when `ADC_CMP_NPCX=n`:
Even with `CONFIG_NO_OPTIMIZATIONS=y`  symbols like `threshold_data_0` are not being included on binaries.